src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -106,15 +134,32 @@ protected override void OnPersist(ApplicationEngine engine)
             // Distribute GAS for committee
 
             var gasPerBlock = GetGasPerBlock(engine.Snapshot);
-            var committee = GetCommitteeMembers(engine.Snapshot).ToArray();
+            var committee = GetCommittee(engine.Snapshot);
             var pubkey = committee[engine.Snapshot.PersistingBlock.Index % ProtocolSettings.Default.CommitteeMembersCount];
             var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
             GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
 
+            // Record the cumulative GAS of the voters
+
+            var voterRewardPerCommittee = gasPerBlock * VoterRewardRatio / 100 / (ProtocolSettings.Default.CommitteeMembersCount + ProtocolSettings.Default.ValidatorsCount);
+            (ECPoint, BigInteger)[] committeeVotes = GetCommitteeVotes(engine.Snapshot);
+            for (var i = 0; i < committeeVotes.Length; i++)
+            {
+                if (committeeVotes[i].Item2 > 0)
+                {
+                    BigInteger voterRewardPerNEO = (i < ProtocolSettings.Default.ValidatorsCount ? 2 : 1) * voterRewardPerCommittee * 10000L / committeeVotes[i].Item2; // Zoom in 10000 times, and the final calculation should be divided 10000L
+                    var enumerator = engine.Snapshot.Storages.Find(CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(committeeVotes[i].Item1).ToArray()).GetEnumerator();
+                    if (enumerator.MoveNext())
+                        voterRewardPerNEO += new BigInteger(enumerator.Current.Value.Value);
+                    var storageKey = CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(committeeVotes[i].Item1).Add(uint.MaxValue - engine.Snapshot.PersistingBlock.Index - 1);
```

Comment:
> why not append just engine.Snapshot.PersistingBlock.Index - 1 to the storageKey?

We need a reverse order, mainly for querying, because we use ascending order in leveldb.

Reply:
We can use `SeekDirection.Backward` for reverse order.

Reply:
Done